grails-app/views/order/_orderItemForm.gsp

```diff
@@ -45,7 +45,7 @@
         <input type="number" id="quantityPerUom" name="quantityPerUom" class="text required" placeholder="${g.message(code: 'default.qtyPerUom.label', default: 'Qty per UoM')}" style="width: 100px"/>
     </td>
     <td class="center middle">
-        <input type="number" id="unitPrice" required name="unitPrice" size="2" class="text required" placeholder="${g.message(code: 'default.pricePerUom.label', default: 'Price per UoM')}" style="width: 100px"/>
+        <input type="text" id="unitPrice" required name="unitPrice" size="2" class="text required" placeholder="${g.message(code: 'default.pricePerUom.label', default: 'Price per UoM')}" style="width: 100px"/>
```

Comment:
What happens now if in fr locale someone will use dot `.` as a decimal point separator? Plus what happens if someone enter text here instead of number?

Reply:
If a user is in a french locale and types in a number with a dot decimal separator like **2.22** it will parse it as **2,00**, so if you switch the language you have to use the appropriate number formatting based on the rules of the given locale.

As for typing in text, it results with a parse exception that is displayed in the error notification like
`{"errorCode":500,"cause":"java.text.ParseException","errorMessage":"Unparseable number: \"dwdw\""}`
so this should probably be fixed.

Do you maybe have a suggestion on what could be done different from the current version ?